addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -130,33 +138,95 @@ private synchronized void onUpdate() {
+        switch (channelUID.getIdWithoutGroup()) {
+            case CHANNEL_TEMP:
+                if (command instanceof RefreshType) {
+                    fritzBox.getTemperature(ain);
+                }
+                break;
+            case CHANNEL_ENERGY:
+                if (command instanceof RefreshType) {
+                    fritzBox.getEnergy(ain);
+                }
+                break;
+            case CHANNEL_POWER:
+                if (command instanceof RefreshType) {
+                    fritzBox.getPower(ain);
+                }
+                break;
+            case CHANNEL_SWITCH:
                 if (command instanceof OnOffType) {
-                    FritzAhaSetSwitchCallback callback = new FritzAhaSetSwitchCallback(fritzBox,
-                            this.getThing().getConfiguration().get(THING_AIN).toString(),
-                            command.equals(OnOffType.ON) ? true : false);
-                    fritzBox.asyncGet(callback);
+                    fritzBox.setSwitch(ain, command.equals(OnOffType.ON) ? true : false);
+                } else if (command instanceof RefreshType) {
+                    fritzBox.getSwitch(ain);
                 }
-            }
-        } else {
-            logger.error("unknown channel uid {}", channelUID);
+                break;
+            case CHANNEL_ACTUALTEMP:
+                if (command instanceof RefreshType) {
+                    // not supported
+                }
+                break;
+            case CHANNEL_SETTEMP:
+                if (command instanceof DecimalType) {
+                    BigDecimal temperature = new BigDecimal(command.toString());
+                    if (temperature.compareTo(HeatingModel.TEMP_MIN) == -1) {
+                        temperature = HeatingModel.TEMP_MIN;
+                    } else if (temperature.compareTo(HeatingModel.TEMP_MAX) == 1) {
+                        temperature = HeatingModel.TEMP_MAX;
+                    }
+                    fritzBox.setSetTemp(ain, temperature.divide(HeatingModel.TEMP_FACTOR));
+                } else if (command instanceof OnOffType) {
+                    BigDecimal temperature = command.equals(OnOffType.ON) ? HeatingModel.TEMP_ON
+                            : HeatingModel.TEMP_OFF;
+                    fritzBox.setSetTemp(ain, temperature);
+                } else if (command instanceof RefreshType) {
+                    fritzBox.getSetTemp(ain);
+                }
+                break;
+            case CHANNEL_ECOTEMP:
+                if (command instanceof RefreshType) {
+                    fritzBox.getEcoTemp(ain);
+                }
+                break;
+            case CHANNEL_COMFORTTEMP:
+                if (command instanceof RefreshType) {
+                    fritzBox.getComfortTemp(ain);
+                }
+                break;
+            case CHANNEL_NEXTCHANGE:
+                if (command instanceof RefreshType) {
```

Comment:
You could use the fall through behavior of the switch (leave this body empty) and since the body of the if is empty you can/should also leave that away.